module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -2066,18 +2088,23 @@ public function newPasswordAction()
                 return $view;
             }
         }
-        // Update password
+        // Set/Reset password
         try {
-            $user = $this->getAuthManager()->updatePassword($this->getRequest());
+            $user = $this->getAuthManager()->newPassword($this->getRequest());
         } catch (AuthException $e) {
             $this->flashMessenger()->addMessage($e->getMessage(), 'error');
             return $view;
         }
         // Update hash to prevent reusing hash
         $this->getAuthManager()->updateUserVerifyHash($user);
-        // Login
+        if ($followUp = $this->followup()->retrieve('url')) {
+            //This exists because the followupURL gets set to Verify which returns
+            //an error message due to trying to check the hash a second time
```

Comment:
Minor style suggestion -- add spaces after `//`:

```suggestion
            // This exists because the followupURL gets set to Verify which returns
            // an error message due to trying to check the hash a second time
```